source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetcherFactory.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Tests.Integration.Support
+{
+    public class TentacleFetcherFactory
+    {
+        private ITentacleFetcher GetBase()
+        {
+            if (PlatformDetection.IsRunningOnNix) return new LinuxTentacleFetcher();
+            return new NugetTentacleFetcher();
+        }
+
+        public ITentacleFetcher Create()
+        {
+            if (TentacleExeFinder.IsRunningInTeamCity())
```

Comment:
I wonder if we should have caching in teamcity? My worry was we might find the tests only work because of the cache but we don't find out until way later.

Then again I think the agents are ephemeral. Thoughts?

Reply:
yeah not sure about this one. As you said, it's not worth caching if the agents themselves get spun up and torn down constantly.

Unless perhaps there's some shared storage the agents have access to? Even then, it's only worthwhile if it's signifcantly faster than re-downloading each time.

Reply:
Ah good point.

I was more concerned about us suddenly depending on the cache being already filled. One of those classic someone updated the build server and now it wont build because the thing it used to get some dependency from doesn't exist any more.

I guess nuget should always be safe. I will make a future PR that adds caching for teamcity in.